### PR TITLE
feat(TDI-41841): Add record headers with tKafkaOutput

### PR DESCRIPTION
**Pull request by Voievidko**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [ ] Unit tests for the Java changes have been added (for bug fixes / features) ?
- [ ] TUJ for the JavaJet changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The new code does not introduce new technical issues

**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-41841

**What is the new behavior?**
Checkbox "Set headers" were added to Advanced settings of tKafkaOutput

**BREAKING CHANGE**

If this PR contains a breaking change, please describe the impact and migration
path for existing applications.
If not please remove this section.

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.components.bigdata/components/tKafkaOutput/tKafkaOutput_java.xml [hunk 1]
```diff
@@ -142,6 +142,30 @@
 				<ITEM NAME="VALUE" />
 			</ITEMS>
 		</PARAMETER>
+
+		<PARAMETER NAME="HEADERS" FIELD="CHECK" NUM_ROW="26"
+			REQUIRED="false" SHOW_IF="((USE_EXISTING_CONNECTION == 'false') AND 
+			((KAFKA_VERSION == 'KAFKA_2_2_1') OR (KAFKA_VERSION == 'KAFKA_1_1_0'))) 
+
+			OR
```
Comment (undx) [inline main/plugins/org.talend.designer.components.bigdata/components/tKafkaOutput/tKafkaOutput_java.xml]: maybe you should reformat the show_if...

File: main/plugins/org.talend.designer.components.bigdata/components/tKafkaOutput/tKafkaOutput_java.xml [hunk 2]
```diff
@@ -142,6 +142,30 @@
...
+
+			((USE_EXISTING_CONNECTION == 'true') AND ((#LINK@NODE.CONNECTION.KAFKA_VERSION == 'KAFKA_2_2_1') OR (#LINK@NODE.CONNECTION.KAFKA_VERSION == 'KAFKA_1_1_0')))">
+			<DEFAULT>false</DEFAULT>
+		</PARAMETER>
+
+		<PARAMETER NAME="KAFKA_HEADERS_PROPERTIES" FIELD="TABLE"
+				SHOW_IF="(HEADERS == 'true') AND (((USE_EXISTING_CONNECTION == 'false') AND 
+			((KAFKA_VERSION == 'KAFKA_2_2_1') OR (KAFKA_VERSION == 'KAFKA_1_1_0'))) 
+
+			OR
```
Comment (undx) [inline main/plugins/org.talend.designer.components.bigdata/components/tKafkaOutput/tKafkaOutput_java.xml]: same

File: main/plugins/org.talend.designer.components.bigdata/components/tKafkaOutput/tKafkaOutput_main.javajet [hunk 3]
```diff
@@ -27,9 +27,34 @@ final String cid = node.getUniqueName();
...
+<%
+        if(isHeaderEnabled && isKafkaVersionSupportHeaders){
+        List<Map<String, String>> headers = (List<Map<String, String>>) ElementParameterParser
+                .getObjectValue(node, "__KAFKA_HEADERS_PROPERTIES__");
 
-<%=cid%>_kafkaProducer.send(new org.apache.kafka.clients.producer.ProducerRecord<byte[], byte[]>(<%=tKafkaOutputUtil.getKafkaTopic()%>, <%=tKafkaOutputUtil.getIncomingConnection().getName()%>.<%=tKafkaOutputUtil.getIncomingColumnName()%>), new org.apache.kafka.clients.producer.Callback() {
+            if(headers != null) {
+	            for (Map<String, String> header : headers) {
+%>
+                    //<%=header.get("VALUE")%>
```
Comment (undx) [inline main/plugins/org.talend.designer.components.bigdata/components/tKafkaOutput/tKafkaOutput_main.javajet]: that was for debugging I think, can remove now?

File: main/plugins/org.talend.designer.components.bigdata/components/tKafkaOutput/tKafkaOutput_main.javajet [hunk 4]
```diff
@@ -27,9 +27,34 @@ final String cid = node.getUniqueName();
...
+  		            record.headers().add(<%=header.get("KEY")%>,
```
Comment (undx) [inline main/plugins/org.talend.designer.components.bigdata/components/tKafkaOutput/tKafkaOutput_main.javajet]: Can KEY be null or empty?

Comment (Voievidko) [reply main/plugins/org.talend.designer.components.bigdata/components/tKafkaOutput/tKafkaOutput_main.javajet]: No, key can't be null.
Key can be empty String.

If key will be null we will receive:
```
java.lang.NullPointerException: Null header keys are not permitted
	at java.util.Objects.requireNonNull(Unknown Source)
	at org.apache.kafka.common.header.internals.RecordHeader.<init>(RecordHeader.java:32)
	at org.apache.kafka.common.header.internals.RecordHeaders.add(RecordHeaders.java:71)
	at local_project.kafka_0_1.Kafka.tRowGenerator_1Process(Kafka.java:1113)
	at local_project.kafka_0_1.Kafka.runJobInTOS(Kafka.java:1804)
	at local_project.kafka_0_1.Kafka.main(Kafka.java:1598)
```

File: main/plugins/org.talend.designer.components.bigdata/components/tKafkaOutput/tKafkaOutput_main.javajet (hunk 3 above)
Comment (Voievidko) [reply main/plugins/org.talend.designer.components.bigdata/components/tKafkaOutput/tKafkaOutput_main.javajet]: done

File: main/plugins/org.talend.designer.components.bigdata/components/tKafkaOutput/tKafkaOutput_java.xml (hunk 2 above)
Comment (Voievidko) [reply main/plugins/org.talend.designer.components.bigdata/components/tKafkaOutput/tKafkaOutput_java.xml]: reformat was done

File: main/plugins/org.talend.designer.components.bigdata/components/tKafkaOutput/tKafkaOutput_java.xml (hunk 1 above)
Comment (Voievidko) [reply main/plugins/org.talend.designer.components.bigdata/components/tKafkaOutput/tKafkaOutput_java.xml]: reformat was done

File: main/plugins/org.talend.designer.components.bigdata/components/tKafkaOutput/tKafkaOutput_main.javajet (hunk 4 above)
Comment (undx) [reply main/plugins/org.talend.designer.components.bigdata/components/tKafkaOutput/tKafkaOutput_main.javajet]: if QA said nothing on this, it OK, thanks!

**Pull request merged**